scripts/compare-deposits/compare-deposits.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"slices"
+	"sort"
+	"time"
+
+	"github.com/nspcc-dev/neo-go/pkg/encoding/address"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/fixedn"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/gas"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/invoker"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/nep17"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+	"github.com/nspcc-dev/neofs-contract/rpc/nns"
+	"github.com/urfave/cli"
+)
+
+func initClient(addr string, name string) (*rpcclient.Client, error) {
+	c, err := rpcclient.New(context.Background(), addr, rpcclient.Options{})
+	if err != nil {
+		return nil, fmt.Errorf("RPC %s: %w", name, err)
+	}
+	err = c.Init()
+	if err != nil {
+		return nil, fmt.Errorf("RPC %s init: %w", name, err)
+	}
+	return c, nil
+}
+
+type deposit struct {
+	from   string
+	amount *big.Int
+	tx     util.Uint256
+}
+
+type mint struct {
+	tx        util.Uint256
+	to        util.Uint160
+	amount    int64
+	depositTx util.Uint256
+	data      []byte
+}
+
+func getSupply(c *rpcclient.Client, h util.Uint160, height uint32) int64 {
+	inv := invoker.NewHistoricAtHeight(height, c, nil)
+	n17 := nep17.NewReader(inv, h)
+	s, err := n17.TotalSupply()
+	if err != nil {
+		return 0
```

Comment:
sure the result will be correct with fallback to zero?

Reply:
It has to be this way because in the beginning of chain's life the is no `balance` contract, so it _always_ returns errors for some blocks. Can be improved with deployment detection, but not worth the trouble, this script worked fine for me on both networks.